lib/kms/file/backend.js

```diff
@@ -0,0 +1,80 @@
+import common from '../common';
+
+export const backend = {
+    /*
+     * Target implementation will be async in some aspects. let mimic it
+     */
+
+    /*
+     * cb(err, masterKeyId: string)
+     */
+    createBucketKey: function createBucketKeyMem(bucketName, log, cb) {
+        process.nextTick(() => {
+            const newKey = common.createDataKey().toString('hex');
+            cb(null, newKey);
+        });
+    },
+
+    /*
+     * cb(err, cipheredDataKey: Buffer)
+     */
+    cipherDataKey: function cipherDataKeyMem(cryptoScheme,
+                                             masterKeyId,
```

Comment:
masterKeyId -- stored with the bucket metadata.  for file backend, the masterKeyId is the actual bucket key rather than the key to retrieve the actual key from a dictionary